# ftp://ftp.kde.org/pub/kde/security_patches/post-3.3.2-kdebase-htmlframes2.patch
#
# Fix for Konqueror Window Injection Vulnerability, see
# http://www.kde.org/info/security/advisory-20041213-1.txt
Index: konq_mainwindow.cc
===================================================================
RCS file: /home/kde/kdebase/konqueror/konq_mainwindow.cc,v
retrieving revision 1.1342.2.14
diff -u -p -r1.1342.2.14 konq_mainwindow.cc
--- kdebase-3.3.2/konqueror/konq_mainwindow.cc	24 Nov 2004 18:29:51 -0000	1.1342.2.14
+++ kdebase-3.3.2/konqueror/konq_mainwindow.cc	12 Dec 2004 15:54:46 -0000
@@ -2252,6 +2252,17 @@ KonqView * KonqMainWindow::childView( KP
     QString viewName = view->viewName();
     kdDebug() << "       - viewName=" << viewName << "   "
               << "frame names:" << view->frameNames().join( "," ) << endl;
+
+    // First look for a hostextension containing this frame name
+    KParts::BrowserHostExtension *ext = KParts::BrowserHostExtension::childObject( view->part() );
+    if ( ext )
+    {
+      ext = ext->findFrameParent(callingPart, name);
+      kdDebug() << "BrowserHostExtension found part " << ext << endl;
+      if (!ext)
+         continue; // Don't use this window
+    }
+
     if ( !viewName.isEmpty() && viewName == name )
     {
       kdDebug() << "found existing view by name: " << view << endl;
@@ -2262,13 +2273,6 @@ KonqView * KonqMainWindow::childView( KP
       return view;
     }
 
-    // First look for a hostextension containing this frame name
-    KParts::BrowserHostExtension *ext = KParts::BrowserHostExtension::childObject( view->part() );
-    if ( ext )
-    {
-      ext = ext->findFrameParent(callingPart, name);
-    }
-
 //    KParts::BrowserHostExtension* ext = KonqView::hostExtension( view->part(), name );
 
     if ( ext )
